Network/DNS/IO.hs

```diff
@@ -36,19 +38,30 @@ import Network.DNS.Types.Internal
 
 ----------------------------------------------------------------
 
+-- | Receive and decode a single 'DNSMessage' from a UDP 'Socket', throwing away
```

Comment:
The original `receive` function is small enough that maintaining two copies is entirely reasonable.  The simpler function avoids unnecessary `sockaddr` allocation when the peer address is not needed.

Instead of reimplementing `receive` as a **wrapper** around `receiveFrom`, please just copy the `receive` function and apply the changes to that copy.  Though it should be safe to call `recvfrom` on a connected (UDP) socket, it is a common usage pattern, and I am also concerned about possible implementation bugs in the kernel (low probability, but best to avoid uncommon code paths).